internal/clients/instance_type_info.go

```diff
@@ -13,15 +13,15 @@ type InstanceTypeInfo struct {
 	RegionalAvailability RegionalTypeAvailability
 }
 
-func (iii *InstanceTypeInfo) InstanceTypesForZone(region, zone string, supported *bool) ([]*InstanceType, error) {
+func (iii *InstanceTypeInfo) InstanceTypesForZone(region, zone string, supported bool) ([]*InstanceType, error) {
 	names, err := iii.RegionalAvailability.NamesForZone(region, zone)
 	if err != nil {
 		return nil, err
 	}
 	result := make([]*InstanceType, 0, len(names))
 	for _, name := range names {
 		rt := iii.RegisteredTypes.Get(name)
-		if supported != nil && *supported != rt.Supported {
+		if supported != rt.Supported {
```

Comment:
I wonder ... could have this been used as "not passing a value => include all"?
Now there is no way to get both `Supported = true` and `Supported = false`.

I'm not sure if we were using it that way, nor I'm saying it was a good use of pointer. Tho now the function changed behaviour, so just making sure it is intended :wink:

Reply:
Yeah, you are correct. Interesting. I think it makes sense as it is, even though I struggled to understand the logic. Thank you for the comment. I will leave out this part for now.

Reply:
Are you changing it back?